gnrc_rpl: proper interface handling

cgundogan opened the pull request:
At DODAG creation, the appropriate interface id will be assigned to the DODAG.
This way, it is not necessary to look up the correct interface id at several places. Furthermore, looking up the interface id like it was done before had some flaws when multiple interfaces were present.

fixes #3970

BytesGalore commented on sys/net/gnrc/routing/rpl/gnrc_rpl.c:
```diff
@@ -117,37 +117,45 @@ gnrc_rpl_instance_t *gnrc_rpl_root_init(uint8_t instance_id, ipv6_addr_t *dodag_
     assert(ipv6 != NULL);
 
+    if (netif) {
+        iface = ((gnrc_netif_hdr_t *) netif->data)->if_pid;
```
please remove whitespace

cgundogan replied:
which whitespace?

BytesGalore commented on sys/net/gnrc/routing/rpl/gnrc_rpl_control_messages.c:
```diff
@@ -83,11 +86,16 @@ void gnrc_rpl_send(gnrc_pktsnip_t *pkt, ipv6_addr_t *src, ipv6_addr_t *dst, ipv6
+    pkt = hdr;
+
+    hdr = gnrc_netif_hdr_build(NULL, 0, NULL, 0);
+    ((gnrc_netif_hdr_t *) hdr->data)->if_pid = iface;
```
please also remove whitespace here

BytesGalore commented on sys/net/gnrc/routing/rpl/gnrc_rpl_control_messages.c:
```diff
@@ -243,11 +251,13 @@ static bool _gnrc_rpl_check_DIS_validity(gnrc_rpl_dis_t *dis, uint16_t len)
 {
     /* TODO handle Solicited Information Option */
     (void) dis;
     (void) len;
```
I guess `dis` and `len` do not require to be "touched" anymore

cgundogan replied:
unrelated to this PR (: but you are right. Will be handled in a follow-up to not bloat the diff

BytesGalore replied:
ok

BytesGalore commented on sys/net/gnrc/routing/rpl/gnrc_rpl_control_messages.c:
```diff
@@ -243,11 +251,13 @@ static bool _gnrc_rpl_check_DIS_validity(gnrc_rpl_dis_t *dis, uint16_t len)
     /* TODO handle Solicited Information Option */
     (void) dis;
     (void) len;
+    (void) iface;
```
please remove the whitespaces between the cast brackets and the casted variable

BytesGalore replied:
ok, I see we have no convention on this (I mixed it up with stuff from work)
Just forget my whitespace terrror

cgundogan replied:
fixed them already (;

BytesGalore commented on sys/net/gnrc/routing/rpl/gnrc_rpl_control_messages.c:
```diff
@@ -937,8 +946,10 @@ static bool _gnrc_rpl_check_DAO_validity(gnrc_rpl_dao_t *dao, uint16_t len)
-void gnrc_rpl_recv_DAO(gnrc_rpl_dao_t *dao, ipv6_addr_t *src, uint16_t len)
+void gnrc_rpl_recv_DAO(gnrc_rpl_dao_t *dao, kernel_pid_t iface, ipv6_addr_t *src, uint16_t len)
 {
+    (void) iface;
```
also here and below

cgundogan commented:
@BytesGalore thanks for the review. Addressed your comments

BytesGalore commented:
I see, sorry for the noise :)

cgundogan commented:
ready to squash?

BytesGalore commented:
yes please squash.
Don't know if it would be cool, but the `rpl` shell command could show the interface of the DODAG.

cgundogan commented:
@BytesGalore that's a nice idea. Squashed and added a new commit adding the interface to the rpl output. I chose to display it at `instance` level, because the `dodag` line is getting quite long.

BytesGalore commented:
nice :+1: ACK
Lets kick the CI

cgundogan commented on sys/net/gnrc/routing/rpl/gnrc_rpl_control_messages.c:
```diff
@@ -536,7 +536,13 @@ void gnrc_rpl_recv_DIO(gnrc_rpl_dio_t *dio, ipv6_addr_t *src, uint16_t len)
-        gnrc_rpl_dodag_init(inst, &dio->dodag_id);
+
+        if (iface == KERNEL_PID_UNDEF) {
+            iface = gnrc_ipv6_netif_find_by_addr(NULL, &ipv6_addr_all_rpl_nodes);
```
@BytesGalore I amended a quick change in this line. I apparently did an error while rebasing, so it was using the static initializer for the all_rpl_nodes address. I changed that to use the global var instead as introduced by a former PR

cgundogan commented:
both CIs agree - GO
